src/Components/Components/src/PersistentComponentState.cs

```diff
@@ -68,11 +68,6 @@ public PersistingComponentStateSubscription RegisterOnPersisting(Func<Task> call
             throw new ArgumentNullException(nameof(key));
         }
 
-        if (key is null)
```

Comment:
[The docs for JsonSerializer.SerializeToUtf8Bytes](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.serializetoutf8bytes) only show an ArgumentNullException when the input's _type_ is null. And a quick [test on C# fiddle](https://dotnetfiddle.net/F3HfdM) shows that a null string doesn't cause an error when serializing. 

`TValue` could also be a nullable type or otherwise intentionally null, in which case we'd probably want to preserve the null through the persist as is.

Let me know if you'd like me to update this to check the instance for null instead of removing the duplicate, but I'm thinking we don't actually want to do that.

Reply:
I think that make sense, thanks for trying it out.